src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -85,14 +90,30 @@ public void periodic() {
 
   }
 
-  //Arcade drive methods.
   public void driveWithJoysticks(double leftThrottle, double rightThrottle, double turn){
     double throttle = rightThrottle - leftThrottle;
-    double forwardSpeed = throttle * 0.8;
-    double turnSpeed = turn * 0.8;
+    double forwardSpeed = throttle * 0.6;
+    double turnSpeed = turn * 0.6;
     diffDrive.arcadeDrive(forwardSpeed, turnSpeed);
   }
 
+  public void driveForward(double driveForwardSpeed){
+    gearShiftSolenoid.set(true);
+    leftMotorGroup.set(driveForwardSpeed);
+    rightMotorGroup.set(driveForwardSpeed);
+  }
+
+  public void chargingStationBalancingWithPID(double kP, double kD, double pitchError){
+    double timeChanges = Timer.getFPGATimestamp() - lastTimestamp;
+    double errorRate = (pitchError - lastError) / timeChanges;
+    double motorOutput = MathUtil.clamp((kP * pitchError + kD * errorRate), -0.4, 0.4);
+    leftMotorGroup.set(motorOutput);
+    rightMotorGroup.set(motorOutput);
+    lastTimestamp = Timer.getFPGATimestamp();
```

Comment:
Are you still using lastTimestamp for anything?  If not, you can get rid of it here.  (And lines 9 and 42.)